
            <!DOCTYPE html>
            <html lang="en">
                <head>
                    <meta charset="UTF-8" />
                    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
                    <title>Schema Evaluator - Individual Analysis</title>
                </head>
                <body style="background-color: #f6f6f6">
                    <noscript> You need to enable JavaScript to run this app. </noscript>
                    <div id="root"></div>
                    <script> const _preComputedDataInd = {"data": {"allele_sizes": [369, 369, 369, 372, 372], "alleles_median": 369, "alleles_mode": 369, "locus_ids": ["GCA-002156805-protein606_1", "GCA-002156805-protein606_S_GCA-015890405.1-PDT000902796.1-genomic.fa_04/10/2022T12:06:03_2", "GCA-002156805-protein606_S_GCA-015891885.1-PDT000902715.1-genomic.fa_04/10/2022T12:06:12_3", "GCA-002156805-protein606_S_GCA-015945895.1-PDT000670959.1-genomic.fa_04/10/2022T12:06:17_4", "GCA-002156805-protein606_S_GCA-018164715.1-PDT001013020.1-genomic.fa_04/10/2022T12:06:24_5"], "nr_alleles": 5, "size_range": "369-372"}, "locus_name": "GCA-002156805-protein606"} </script>
                    <script> const _exceptions = [] </script>
                    <script> const _cdsDf = {"Alleles not multiple of 3": 0, "Alleles shorter than 201 nucleotides": 0, "Alleles w/ >1 stop codons": 0, "Alleles wo/ Start/Stop Codon": 0, "CDS": 5, "Gene": "GCA-002156805-protein606.fasta", "Missing Allele IDs": ["None"], "Number of alleles": 5, "Total Invalid Alleles": 0, "coding_strand": "Not provided", "contig": "Not provided", "genome": "Not provided", "name": "Not provided", "proteome_bsr": "Not provided", "proteome_gene_name": "Not provided", "proteome_id": "Not provided", "proteome_product": "Not provided", "proteome_species": "Not provided", "start": "Not provided", "stop": "Not provided", "url": "Not provided"} </script>
                    <script> const _msaData = {"sequences": [{"name": "1", "sequence": "MQAKYLPEDSLAYLAASARVAIDAATEAIQDVNGQCPEFEDLPGNLQDAIYILDGVRDAIRGEAMAHNVHRATHYSNGYPIRIHYKGAAITDPSGVRYRPDTITTMHPSGTTDAPIPPQDIVA"}, {"name": "2", "sequence": "-MTNYTPQDSLAYLAASARVAIDAATEAIQDVNGQCPEFEDLPGNLQDAIYILDGVRDAIRKEAMAHNVHRATHYSNGYPIRIHYKDAAITDPSGVRYRPDTITAMHPNGTPDAPIPPQDIVA"}, {"name": "3", "sequence": "-MTDYTPQDSLAYLADRVRCSLDAAIEAIQDVNGQCPEFEDLPGNLQDAIYILDGVRDAIRKEAMAHNVHRATHYSNGYPIRIHYKGAAITDPSGARYRPDTITTMHPSGKSDSPIPPQDIVA"}, {"name": "4", "sequence": "MLAKYLPQDSLAYLAASARVAIDAATEAIQDVNGQCPEFEDLPGNLQDAIYILDGVRDAIRSEAMAHNVHRATHYSNGYPVRVHYKGTAVTDDEGRCYRPDTVTAMHPSGTTDAPIPPQDIVA"}, {"name": "5", "sequence": "-MTDYSPQDSLAYLAASARVAIDAATEAIQDVNGQCPEFEDLPGNLQDAIYILDGVRDAIRGEAMAHNVHRATHYSNGYPVRVHYKGTAVTDPSGVRYRPDTITAMHPSGKTDAPIPPQDIVA"}]} </script>
                    <script> const _phyloData = {"phylo_data": "((((\n1_1\n:0.08570,\n5_5\n:0.08570):0.01048,\n4_4\n:0.09618):0.05791,\n2_2\n:0.15409):0.07642,\n3_3\n:0.23052);\n"} </script>
                    <script> const _minLen = 201 </script>
                    <script> const _fasta = `>1
MQAKYLPEDSLAYLAASARVAIDAATEAIQDVNGQCPEFEDLPGNLQDAIYILDGVRDAI
RGEAMAHNVHRATHYSNGYPIRIHYKGAAITDPSGVRYRPDTITTMHPSGTTDAPIPPQD
IVA
>2
-MTNYTPQDSLAYLAASARVAIDAATEAIQDVNGQCPEFEDLPGNLQDAIYILDGVRDAI
RKEAMAHNVHRATHYSNGYPIRIHYKDAAITDPSGVRYRPDTITAMHPNGTPDAPIPPQD
IVA
>3
-MTDYTPQDSLAYLADRVRCSLDAAIEAIQDVNGQCPEFEDLPGNLQDAIYILDGVRDAI
RKEAMAHNVHRATHYSNGYPIRIHYKGAAITDPSGARYRPDTITTMHPSGKSDSPIPPQD
IVA
>4
MLAKYLPQDSLAYLAASARVAIDAATEAIQDVNGQCPEFEDLPGNLQDAIYILDGVRDAI
RSEAMAHNVHRATHYSNGYPVRVHYKGTAVTDDEGRCYRPDTVTAMHPSGTTDAPIPPQD
IVA
>5
-MTDYSPQDSLAYLAASARVAIDAATEAIQDVNGQCPEFEDLPGNLQDAIYILDGVRDAI
RGEAMAHNVHRATHYSNGYPVRVHYKGTAVTDPSGVRYRPDTITAMHPSGKTDAPIPPQD
IVA
` </script>
                    <script src="./main_ind.js"></script>
                </body>
            </html>
            